imap_processing/lo/tests/test_science_direct_events.py

```diff
@@ -0,0 +1,290 @@
+import bitstring
+
+from imap_processing.lo.l0.science_direct_events import ScienceDirectEvents
+
+# TODO: These unit tests don't call the public function for the ScienceDirectEvents
+# class and are also only able to run if the contects of __init__ are commented out
+# because I currently don't have any compressed DE data. As a workaround, I am
+# directly setting the DATA attribute and calling and testing the private methods
+# as needed. These tests will be updated when data becomes available. When that happens
+# the need for the bitstring import will also go away.
+
+
+def test_find_decompression_case():
+    # Arrange
+    de = ScienceDirectEvents("fake_packet", "0", "fakepacketname")
+    de.DATA = bitstring.Bits(bin="000100010101")
+    case_number_expected = 1
+
+    # Act
+    de._find_decompression_case()
+    case_number_true = de.case_number
+
+    # Assert
+    assert case_number_true == case_number_expected
+
+
+def test_find_tof_decoder_for_case():
+    # Arrange
+    de = ScienceDirectEvents("fake_packet", "0", "fakepacketname")
+    de.DATA = bitstring.Bits(bin="000100010101")
+    tof_decoder_expected = {
+        "ENERGY": 3,
+        "POS": 0,
+        "TOF0": 10,
+        "TOF1": 9,
+        "TOF2": 9,
+        "TOF3": 0,
+        "CKSM": 0,
+        "TIME": 12,
+    }
+    de._find_decompression_case()
+
+    # Act
+    de._find_tof_decoder_for_case()
+    tof_decoder_true = de.tof_decoder
+
+    # Assert
+    assert tof_decoder_true == tof_decoder_expected
+
+
+def test_read_tof_calculation_table():
+    # Arrange
+    de = ScienceDirectEvents("fake_packet", "0", "fakepacketname")
+    de.DATA = bitstring.Bits(bin="000100010101")
+    de._find_decompression_case()
+    binary_strings_expected = {
+        "TIME": bitstring.Bits(bin="0000111111111111"),
```

Comment:
Is this table always the same? could it be a class instead of a dict? (can you tell I don't like python dicts... haha)

Reply:
haha whatever is going on between you and python dictionaries seems personal. Maybe I haven't thought about it enough, but the problem is sometimes those values don't exist depending on the case number. And different case numbers have different combinations of those values that don't exist. So I'd need to have a bunch of logic in there to pick with of those values are populated based on the case number.

Reply:
Yeah, makes sense to me. And I think my real problem with python dictionaries is actually because of Pythons type system. I don't mind hashtables in other languages! But they can get gnarly with ducktyping 🦆